src/service/safety.ts

```diff
@@ -71,6 +73,29 @@ export class SafetyCalculator {
     private computeQtyLimit = () => {
         var settings = this._repo.latest;
 
+        var buyPing = 0;
+        var sellPong = 0;
+        var buyPq = 0;
+        var sellPq = 0;
+        var _buyPq = 0;
+        var _sellPq = 0;
+        for (var ti = this._broker._trades.length - 1; ti > -1; ti--) {
+          if (this._broker._trades[ti].side == Models.Side.Bid && buyPq<settings.size) {
```

Comment:
Looking at this now, I think I'll make a note to one day make `_trades` part of a TradeBroker interface. I never intended for it to be public (but have used it in a public way throughout the code base)